fireapp/lib/firebase/firebase_api.dart

```diff
@@ -0,0 +1,63 @@
+import 'package:firebase_messaging/firebase_messaging.dart';
+import 'package:flutter_local_notifications/flutter_local_notifications.dart';
+
+class FirebaseApi {
+  final _firebaseMessaging = FirebaseMessaging.instance;
+  final _localNotifications = FlutterLocalNotificationsPlugin();
+
+  Future<void> initNotifications() async {
+    await _firebaseMessaging.requestPermission();
+    final fcmToken = await _firebaseMessaging.getToken();
+    print('Token: $fcmToken');
+
+    initPushNotifications();
+    initLocalNotifications();
+  }
+
+  Future<void> initPushNotifications() async {
+    await FirebaseMessaging.instance.setForegroundNotificationPresentationOptions(
+      alert: true,
+      badge: true,
+      sound: true,
+    );
+
+    FirebaseMessaging.onMessage.listen((message) {
+      final notification = message.notification;
+      if (notification == null) return;
+
+      _localNotifications.show(
+        notification.hashCode,
+        notification.title,
+        notification.body,
+        NotificationDetails(
+          android: AndroidNotificationDetails(
+            'high_importance_channel',
+            'High Importance Notifications',
+            importance: Importance.max,
+          ),
+          iOS: DarwinNotificationDetails(),
+        ),
+        payload: message.data['route'],
```

Comment:
Not really a request for change, more just a question. Has this payload been coordinated with the backend team?

Reply:
Ah yeah - I forget to remove that. That is needed if we want to route the user to some view within the app. Do you think we need this? 

I haven't told the backend team about this yet.

Reply:
It would be a good feature to allow us to route to pages from notifications. I would coordinate with them on something

Reply:
Ah okay. Do I need to create another pull request for it to be implemented? Or should I wait for coordination w/ the backend team first before implementing this? (I think it's only several lines - but the payload obviously has to be agreed w/ the backend team).

Reply:
Let's just pull it out for now, you can revert the change once it's been agreed upon